discord/ext/commands/core.py

```diff
@@ -271,74 +320,85 @@ def __init__(self, func, **kwargs):
-        self.extras = kwargs.get('extras', {})
+        self.brief: Optional[str] = kwargs.get('brief')
+        self.usage: Optional[str] = kwargs.get('usage')
+        self.rest_is_raw: bool = kwargs.get('rest_is_raw', False)
+        self.aliases: Union[List[str], Tuple[str]] = kwargs.get('aliases', [])
+        self.extras: Dict[str, Any] = kwargs.get('extras', {})
 
         if not isinstance(self.aliases, (list, tuple)):
             raise TypeError("Aliases of a command must be a list or a tuple of strings.")
 
-        self.description = inspect.cleandoc(kwargs.get('description', ''))
-        self.hidden = kwargs.get('hidden', False)
+        self.description: str = inspect.cleandoc(kwargs.get('description', ''))
+        self.hidden: bool = kwargs.get('hidden', False)
 
         try:
             checks = func.__commands_checks__
             checks.reverse()
         except AttributeError:
             checks = kwargs.get('checks', [])
-        finally:
-            self.checks = checks
+
+        self.checks: List[Check] = checks
 
         try:
             cooldown = func.__commands_cooldown__
         except AttributeError:
             cooldown = kwargs.get('cooldown')
-        finally:
-            if cooldown is None:
-                self._buckets = CooldownMapping(cooldown, BucketType.default)
-            elif isinstance(cooldown, CooldownMapping):
-                self._buckets = cooldown
+        
+        if cooldown is None:
+            buckets = CooldownMapping(cooldown, BucketType.default)
+        elif isinstance(cooldown, CooldownMapping):
+            buckets = cooldown
+        else:
+            raise TypeError("Cooldown must be a an instance of CooldownMapping or None.")
+        self._buckets: CooldownMapping = buckets
 
         try:
             max_concurrency = func.__commands_max_concurrency__
         except AttributeError:
             max_concurrency = kwargs.get('max_concurrency')
-        finally:
-            self._max_concurrency = max_concurrency
 
-        self.require_var_positional = kwargs.get('require_var_positional', False)
-        self.ignore_extra = kwargs.get('ignore_extra', True)
-        self.cooldown_after_parsing = kwargs.get('cooldown_after_parsing', False)
-        self.cog = None
+        self._max_concurrency: Optional[MaxConcurrency] = max_concurrency
+
+        self.require_var_positional: bool = kwargs.get('require_var_positional', False)
+        self.ignore_extra: bool = kwargs.get('ignore_extra', True)
+        self.cooldown_after_parsing: bool = kwargs.get('cooldown_after_parsing', False)
+        self.cog: Optional[CT] = None
 
         # bandaid for the fact that sometimes parent can be the bot instance
         parent = kwargs.get('parent')
-        self.parent = parent if isinstance(parent, _BaseCommand) else None
+        self.parent: Optional[Command] = parent if isinstance(parent, _BaseCommand) else None  # type: ignore
 
```

Comment:
Should this be `GroupMixin` or `Group`?
Technically this could be the `Bot`.

I've left as `Command` for now.

Reply:
It should probably be `GroupMixin`